pipelines/infrastructure/cortextool/cortexttool.yaml

```diff
@@ -1,72 +1,138 @@
+          cortextool alertmanager load gitrepo-alertmanager-config/alertmanager.yaml
+        path: sh
+    task: push-alertmanager-config
+- build_log_retention:
+    days: 90
+  name: sync-alertmanager-config-to-qa
+  plan:
+  - get: gitrepo-alertmanager-config
+    passed:
+    - sync-alertmanager-config-to-ci
+    trigger: true
+  - config:
       image_resource:
-        type: registry-image
+        name: ""
         source:
           repository: grafana/cortex-tools
           tag: v0.10.7
+        type: registry-image
+      inputs:
+      - name: gitrepo-alertmanager-config
+      platform: linux
       run:
+        args:
+        - -exc
+        - |
+          export CORTEX_ADDRESS="((cortextool.cortex-amconfig-api-address))"
+          export CORTEX_API_USER="((cortextool.cortex-amconfig-api-user-qa))"
+          export CORTEX_API_KEY="((cortextool.cortex-api-key-qa))"
+          export CORTEX_TENANT_ID="((cortextool.cortex-amconfig-api-user-qa))"
+          sed -i -e 's/%% OPS_TEAM_OPS_GENIE_API_KEY %%/((cortextool.ops-team-ops-genie-api-key))/' gitrepo-alertmanager-config/alertmanager.yaml
+          sed -i -e 's/%% TESTING_OPS_GENIE_API_KEY %%/((cortextool.testing-ops-genie-api-key))/' gitrepo-alertmanager-config/alertmanager.yaml
+          sed -i -e 's/%% ENVIRONMENT_NAME %%/QA/' gitrepo-alertmanager-config/alertmanager.yaml
+          cortextool alertmanager load gitrepo-alertmanager-config/alertmanager.yaml
         path: sh
+    task: push-alertmanager-config
+- build_log_retention:
+    days: 90
+  name: sync-alertmanager-config-to-production
+  plan:
+  - get: gitrepo-alertmanager-config
+    passed:
+    - sync-alertmanager-config-to-qa
+    trigger: true
+  - config:
+      image_resource:
+        name: ""
+        source:
+          repository: grafana/cortex-tools
+          tag: v0.10.7
+        type: registry-image
+      inputs:
+      - name: gitrepo-alertmanager-config
+      platform: linux
+      run:
         args:
         - -exc
         - |
-          cortextool rules lint gitrepo-cortex-rules/cortex-rules/*.yaml
-          cp -Rv gitrepo-cortex-rules/. changed-repo
-  - task: commit-linted-cortex-rules
-    config:
+          export CORTEX_ADDRESS="((cortextool.cortex-amconfig-api-address))"
+          export CORTEX_API_USER="((cortextool.cortex-amconfig-api-user-production))"
+          export CORTEX_API_KEY="((cortextool.cortex-api-key-qa))"
+          export CORTEX_TENANT_ID="((cortextool.cortex-amconfig-api-user-production))"
+          sed -i -e 's/%% OPS_TEAM_OPS_GENIE_API_KEY %%/((cortextool.ops-team-ops-genie-api-key))/' gitrepo-alertmanager-config/alertmanager.yaml
+          sed -i -e 's/%% TESTING_OPS_GENIE_API_KEY %%/((cortextool.testing-ops-genie-api-key))/' gitrepo-alertmanager-config/alertmanager.yaml
+          sed -i -e 's/%% ENVIRONMENT_NAME %%/Production/' gitrepo-alertmanager-config/alertmanager.yaml
+          cortextool alertmanager load gitrepo-alertmanager-config/alertmanager.yaml
```

Comment:
It should be possible to turn this into a script in the repo that you can run which uses environment variables that you can set via params to the task step so that you don't have to do as much copy/paste across the different steps.